File: Classes/Util/ARIntercomProvider.m
@import Intercom;
#import "ARIntercomProvider.h"


@implementation ARIntercomProvider

- (instancetype)initWithWithAppID:(NSString *)identifier apiKey:(NSString *)apiKey
{
    NSAssert([Intercom class], @"Intercom is not included");

    [Intercom setApiKey:apiKey forAppId:identifier];

    return [super init];
}


- (void)setUserProperty:(NSString *)property toValue:(NSString *)value
{
    [Intercom updateUserWithAttributes:@{ @"custom_attributes" : @{property : value} }];
}


@end

File: Classes/Util/ARIntercomProvider.swift
import ARAnalytics
import Intercom

class ARIntercomProvider: ARAnalyticalProvider {

    init(apiKey: String, appID: String) {
        Intercom.setApiKey(apiKey, forAppId: appID)
        super.init()
    }

    override func setUserProperty(property: String!, toValue value: String!) {
        Intercom.updateUserWithAttributes(["custom_attributes" : [property:value]])
    }
}
